py-rse/bash-advanced.Rmd

```diff
@@ -81,94 +80,91 @@ For example,
 we can count how many lines of output there are by putting our script in a pipeline:
 
 ```shell
-$ bash years.sh | wc -l
+$ bash book_summary.sh | wc -l
 ```
 
 ```text
-    8314
+    11
 ```
 
-What if we want to remove duplicates from the output?
-The command `uniq` will do what we want,
+What if we want to extract the name of the author?
+The command `grep` searches for every line that contains a pattern and prints the matching lines (we will learn more about `grep` later).
 so let's edit our script,
 add it,
 look at our changes,
 and then run the modified script:
 
 ```shell
-$ nano years.sh
-$ cat years.sh
+$ nano book_summary.sh
+$ cat book_summary.sh
 ```
 
 ```text
-cut -d , -f 2 cleaned/bellambi_temp.csv | uniq
+head -n 19 data/moby_dick.txt | tail -n 11 | grep Author
 ```
 
 ```shell
-$ bash years.sh
+$ bash book_summary.sh
 ```
 
 ```text
-Year
-1997
-1998
-1999
-...one line per year...
-2017
-2018
-2019
+Author: Herman Melville
 ```
 
 Once again,
 we can pipe the output of our script into other commands
 just as we would pipe the output from any other program:
 
 ```shell
-$ bash years.sh | wc -l
+$ bash book_summary.sh | wc -w
 ```
 
 ```text
-      24
+      3
 ```
 
 ## How can I make my scripts more versatile? {#py-rse-bash-advanced-params}
 
 Creating a list of distinct years in a single specific data file isn't all that useful.
 What we really want is a way to get the years from any of our files.
-Let's edit `years.sh` again and replace `cleaned/bellambi_temp.csv`
+Let's edit `book_summary.sh` again and replace `data/moby_dick.csv`
 with a special variable `$1`.
 Once our change is made,
 `years.sh` should contain:
 
 ```text
-cut -d , -f 2 $1 | uniq
+head -n 19 $1 | tail -n 11 | grep Author
```

Comment:
There are some references to `years.sh` in the paragraph above that need to be removed.